aliases.go

```diff
@@ -78,3 +104,57 @@ func (client *Client) DeleteAlias(input AliasDeleteInput) error {
 	err := client.Mutate(&m, v, WithName("AliasDelete"))
 	return HandleErrors(err, m.Payload.Errors)
 }
+
+// ReconcileAliases manages aliases API operations for AliasOwnerInterface implementations
+//
+// Aliases not in 'aliasesWanted' will be deleted, new tags from 'aliasesWanted' will be created. Reconciled aliases are returned.
+func (client *Client) ReconcileAliases(resourceType AliasOwnerInterface, aliasesWanted []string) ([]string, error) {
+	var deleteAliasFunc func(string) error
+	var existingAliases []string
+	var resourceId ID
+
+	switch resource := resourceType.(type) {
+	case *Domain:
+		deleteAliasFunc = client.DeleteDomainAlias
+		existingAliases = resource.ManagedAliases
+		resourceId = resource.Id
+	case *InfrastructureResource:
+		deleteAliasFunc = client.DeleteInfraAlias
+		existingAliases = resource.Aliases
+		resourceId = ID(resource.Id)
+	case *Scorecard:
+		deleteAliasFunc = client.DeleteScorecardAlias
+		existingAliases = resource.Aliases
+		resourceId = resource.Id
+	case *Service:
+		deleteAliasFunc = client.DeleteServiceAlias
+		existingAliases = resource.ManagedAliases
+		resourceId = resource.Id
+	case *System:
+		deleteAliasFunc = client.DeleteSystemAlias
+		existingAliases = resource.Aliases
+		resourceId = resource.Id
+	case *Team:
+		deleteAliasFunc = client.DeleteTeamAlias
+		existingAliases = resource.ManagedAliases
+		resourceId = resource.Id
+	}
+
+	// delete aliases found in resource but not listed in aliasesWanted
+	for _, alias := range existingAliases {
+		if !slices.Contains(aliasesWanted, alias) {
+			if err := deleteAliasFunc(alias); err != nil {
+				return []string{}, err
+			}
+		}
+	}
+
+	newServiceAliases := []string{}
+	for _, aliasWanted := range aliasesWanted {
+		if !slices.Contains(existingAliases, aliasWanted) {
+			newServiceAliases = append(newServiceAliases, aliasWanted)
+		}
+	}
```

Comment:
Then you can extract out this functionality to a more simple function that takes in `currentAliases`, `desiredAliases` and return `aliasesToCreate` and `aliasesToDelete`

So you do something like
```
aliasesToCreate, aliasesToDelete := reconcileAliases(domain.Aliases, []string{"one","three","five"})
```

This also makes it so that `reconcileAliases` is easily unit testable with different string slices as input to ensure the reconcile logic is correct.